### PR TITLE
MM-10658 Change config fields to pointers

### PR DESCRIPTION
**Pull request by jwilander**
#### Summary
Based on the PR #8898 submitted by @chclaus but with the EE changes so that the build will run.

Enterprise PR: https://github.com/mattermost/enterprise/pull/310

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-10658

### CODE REVIEW

**Comment by jwilander** [comment]
@chclaus I'm hopefully going to get back to this soon to rebase it and figure out why some of the tests were failing. If you have time feel free to take a stab at it :)

**Comment by cpanato** [comment]
@jwilander can you please rebase and resolve the conflicts?
 thanks!

**Comment by jwilander** [comment]
I have an item to revisit this next week

**Comment by lieut-data** [comment]
I'm taking over the last leg of this PR as part of the planned configuration rework: I've updated the PR with the latest master and updated the new code added since then. Just waiting for build stabilization.

File: model/config.go [hunk 1]
```diff
@@ -720,8 +783,16 @@ func (s *SqlSettings) SetDefaults() {
 		s.DataSource = NewString(SQL_SETTINGS_DEFAULT_DATA_SOURCE)
 	}
 
-	if len(s.AtRestEncryptKey) == 0 {
-		s.AtRestEncryptKey = NewRandomString(32)
+	if s.DataSourceReplicas == nil {
+		s.DataSourceReplicas = []string{}
```
Comment (grundleborg) [inline model/config.go]: Probably a naive question, but is there a reason we aren't making these list-type variables into pointers too?

Comment (grundleborg) [reply model/config.go]: Makes perfect sense to me, thanks.

**Pull request merged**